src/components/shareable-url-modal/main-view/main-view.js

```diff
@@ -0,0 +1,190 @@
+/* eslint-disable camelcase */
+import React from 'react';
+import classnames from 'classnames';
+import Dropdown from '../../ui/dropdown';
+import Button from '../../ui/button';
+import IconButton from '../../ui/icon-button';
+import InfoIcon from '../../icons/info';
+import Input from '../../ui/input';
+import MenuOption from '../../ui/menu-option';
+import Toggle from '../../ui/toggle';
+import {
+  hostingPlatforms,
+  KEDRO_VIZ_PUBLISH_AWS_DOCS_URL,
+  KEDRO_VIZ_PUBLISH_AZURE_DOCS_URL,
+  KEDRO_VIZ_PUBLISH_GCP_DOCS_URL,
+  KEDRO_VIZ_PUBLISH_DOCS_URL,
+} from '../../../config';
+
+const renderTextContent = (toggleValue, setTogleValue) => {
```

Comment:
(toggleValue, setTogleValue) - if this is only for previews, can we rename to (isPreviewEnabled, setIsPreviewEnabled) ?